### PR TITLE
Consolidate extraction of tokens from annotated string field value.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchlib/src/vespa/searchlib/util/token_extractor.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+bool
+TokenExtractor::extract(bool allow_zero_length_tokens, std::vector<SpanTerm>& terms, const document::StringFieldValue::SpanTrees& trees)
+{
+    auto tree = StringFieldValue::findTree(trees, SPANTREE_NAME);
+    if (tree == nullptr) {
+        return false;
+    }
+    terms.clear();
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/util/token_extractor.cpp:70]: Consider moving this responsibility to the caller of the function.

**Review by geirst — DISMISSED** [review]
👍

File: searchlib/src/vespa/searchlib/util/token_extractor.cpp (hunk 1 above)
Comment (toregge) [reply searchlib/src/vespa/searchlib/util/token_extractor.cpp]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**